### ISSUE TITLE
[Server]: Nestjs Error when entity has a field referencing itself

### ISSUE THREAD

**Issue by Tug** [amplication/amplication#1126]
**Describe the bug**
When creating a self relation in an entity, the generated app will fail to start with the following error in the logs
```
server_1   | (node:1) UnhandledPromiseRejectionWarning: Error: "Model" cannot be found in the registry (ModelResolver#model). This is often caused by missing argument name in the method signature. A potential fix: change @Args() to @Args('argumentName').
server_1   |     at /app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/args.factory.js:30:27
server_1   |     at Array.forEach (<anonymous>)
server_1   |     at ArgsFactory.create (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/args.factory.js:17:14)
server_1   |     at /app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:40:40
server_1   |     at Array.forEach (<anonymous>)
server_1   |     at RootTypeFactory.generateFields (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:34:14)
server_1   |     at fieldsFactory (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:18:101)
server_1   |     at RootTypeFactory.create (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/root-type.factory.js:27:21)
server_1   |     at QueryTypeFactory.create (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/factories/query-type.factory.js:15:37)
server_1   |     at GraphQLSchemaFactory.<anonymous> (/app/server/node_modules/@nestjs/graphql/dist/schema-builder/graphql-schema.factory.js:42:46)
server_1   | (Use `node --trace-warnings ...` to show where the warning was created)
server_1   | (node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
server_1   | (node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

On build time, we can notice another related error when docker runs `nest build`:
```
Creating an optimized production build...
src/model/model.resolver.ts:53:9 - error TS2393: Duplicate function implementation.

53   async model(
           ~~~~~
src/model/model.resolver.ts:199:9 - error TS2393: Duplicate function implementation.

199   async model(
            ~~~~~

Found 2 error(s).
```

Indeed since the Model entity has a property with the same type.

**To Reproduce**
1. Go to Entities and open any existing entity (let's call it `Model`)
2. In `Entity Fields` enter any text and click `Add field`
3. A right pane with the field config should open, in `Data Type`, select `Relation to another entity`
4. In `Related Entity` select `Model`
5. Commit changes, download code and run the project
6. You should see the error

**Expected behavior**
Either the UI should prevent me from creating self references or it should work fine with it.

**Screenshots**
![image](https://user-images.githubusercontent.com/230230/104503112-ed778380-55e0-11eb-9b87-3a2f8781e1cf.png)

**Comment by yuval-hazaz** [comment]
Thanks, @Tug for reporting. 
There is an open PR handling relation fields #1080 
We will make sure to verify a fix for this issue is included

**Comment by yuval-hazaz** [comment]
resolved in #1080
